projects/plugins/boost/app/assets/src/js/features/super-cache-info/super-cache-info.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import { DataSyncProvider, useDataSync } from '@automattic/jetpack-react-data-sync-client';
+import { __, sprintf } from '@wordpress/i18n';
+import { Notice, Button } from '@automattic/jetpack-components';
+import {
+	measureSuperCacheSaving,
+	isSuperCachePluginActive,
+	isSuperCacheEnabled,
+} from './lib/utils/measure';
+import { z } from 'zod';
+
+const SuperCacheInfo = () => {
+	const [ state, setState ] = useState( { status: 'idle' } ); // idle, testing, error, complete
+
+	const { useQuery, useMutation } = useDataSync(
+		'jetpack_boost_ds',
+		'super_cache_notice_disabled',
+		z.boolean()
+	);
+	const { data: isNoticeDismissed } = useQuery();
+	const { mutate: setNoticeDismissed } = useMutation();
+
+	const runTest = async () => {
+		setState( { status: 'testing' } );
+		try {
+			const saving = await measureSuperCacheSaving();
+			setState( { status: 'complete', saving } );
+		} catch ( error ) {
+			setState( { status: 'error', error } );
+		}
+	};
+
+	const navToSuperCacheSettings = () => {
+		window.location.href = './options-general.php?page=wpsupercache';
+	};
+
+	if ( isSuperCacheEnabled() ) {
+		if ( state.status === 'testing' ) {
+			return (
+				<Notice
+					level="info"
+					title={ __( 'Measuring Super Cache Speed', 'jetpack-boost' ) }
+					hideCloseButton={ true }
+					actions={ [
+						<Button key="running" isPrimary isLoading>
+							{ __( 'Run test', 'jetpack-boost' ) }
+						</Button>,
+					] }
+				>
+					<p>{ __( 'Jetpack Boost is testing the speed of your cache.', 'jetpack-boost' ) }</p>
+				</Notice>
+			);
+		} else if ( state.status === 'idle' ) {
+			return (
+				<Notice
+					level="info"
+					title={ __( 'Super Cache detected', 'jetpack-boost' ) }
+					hideCloseButton={ true }
+					actions={ [
+						<Button
+							key="start"
+							isPrimary
+							onClick={ () => {
+								runTest();
+							} }
+						>
+							{ __( 'Run test', 'jetpack-boost' ) }
+						</Button>,
+					] }
+				>
+					<p>{ __( 'Find out how much difference it makes for your users.', 'jetpack-boost' ) }</p>
+				</Notice>
+			);
+		} else if ( state.status === 'complete' ) {
+			return (
+				<Notice
+					level="success"
+					title={ __( 'Super Cache Speed', 'jetpack-boost' ) }
+					hideCloseButton={ true }
+					actions={ [
+						<Button
+							key="start"
+							isPrimary
+							onClick={ () => {
+								runTest();
+							} }
+						>
+							{ __( 'Re-run test', 'jetpack-boost' ) }
+						</Button>,
+					] }
+				>
+					<p>
+						{ sprintf(
+							// translators: %d refers to the number of milliseconds users are saving by using Super Cache.
+							__( 'Super Cache is saving your visitors about %d ms', 'jetpack-boost' ),
+							state.saving
+						) }
+					</p>
+				</Notice>
+			);
+		}
+	} else {
+		if ( isNoticeDismissed ) {
+			return null;
+		}
+
+		if ( isSuperCachePluginActive() ) {
```

Comment:
I've been trying to simplify these if else returns and I'm not seeing a good way of doing it.

The only thing that kept popping up in my attempts was moving this to the top, since if the plugin isn't active, there's no point in checking if cache is enabled or not.

For example:

```js
// If Super Cache plugin is not active, do nothing.
if ( ! isSuperCachePluginActive() ) {
	return null;
}

// If cache is not enabled and notice is not dismissed, show notice.
if ( ! isSuperCacheEnabled() && ! isNoticeDismissed ) {
	// caching disabled...
}

// code for when caching is enabled
```

Reply:
Oh! Lightbulb moment.

See if this is easier to read:

```js
	if ( ! isSuperCachePluginActive() ) {
		return null;
	}

	if ( isNoticeDismissed ) {
		return null;
	}

	if ( ! isSuperCacheEnabled() ) {
		return (
			<Notice
				level="warning"
				title={ __( 'Super Cache is installed but not enabled', 'jetpack-boost' ) }
				actions={ [
					<Button
						key="start"
						isPrimary
						onClick={ () => {
							navToSuperCacheSettings();
						} }
					>
						{ __( 'Set up', 'jetpack-boost' ) }
					</Button>,
				] }
				hideCloseButton={ false }
				onClose={ () => {
					setNoticeDismissed( true );
				} }
			>
				{ __( 'Enable Super Cache to speed your site up further.', 'jetpack-boost' ) }
			</Notice>
		);
	}

	return (
		<>
			{ state.status === 'testing' && (
				<Notice
					level="info"
					title={ __( 'Measuring Super Cache Speed', 'jetpack-boost' ) }
					hideCloseButton={ true }
					actions={ [
						<Button key="running" isPrimary isLoading>
							{ __( 'Run test', 'jetpack-boost' ) }
						</Button>,
					] }
				>
					<p>{ __( 'Jetpack Boost is testing the speed of your cache.', 'jetpack-boost' ) }</p>
				</Notice>
			) }
			{ state.status === 'idle' && (
				<Notice
					level="info"
					title={ __( 'Super Cache detected', 'jetpack-boost' ) }
					hideCloseButton={ true }
					actions={ [
						<Button
							key="start"
							isPrimary
							onClick={ () => {
								runTest();
							} }
						>
							{ __( 'Run test', 'jetpack-boost' ) }
						</Button>,
					] }
				>
					<p>{ __( 'Find out how much difference it makes for your users.', 'jetpack-boost' ) }</p>
				</Notice>
			) }
			{ state.status === 'complete' && (
				<Notice
					level="success"
					title={ __( 'Super Cache Speed', 'jetpack-boost' ) }
					hideCloseButton={ true }
					actions={ [
						<Button
							key="start"
							isPrimary
							onClick={ () => {
								runTest();
							} }
						>
							{ __( 'Re-run test', 'jetpack-boost' ) }
						</Button>,
					] }
				>
					<p>
						{ sprintf(
							// translators: %d refers to the number of milliseconds users are saving by using Super Cache.
							__( 'Super Cache is saving your visitors about %d ms', 'jetpack-boost' ),
							state.saving
						) }
					</p>
				</Notice>
			) }
		</>
	);
```

Reply:
Thanks, that works better. :)